### PR TITLE
WIKI-1397 fix the problem with special characters in attachment name

### PR DESCRIPTION
**Pull request by trongtt**

### CODE REVIEW

File: wiki-service/src/main/java/org/exoplatform/wiki/utils/Utils.java [hunk 1]
```diff
@@ -80,7 +80,7 @@
   
   private static final String ILLEGAL_SEARCH_CHARACTERS= "\\!^()+{}[]:-\"";
   
-  private static final String ILLEGAL_NAME_CHARACTERS = "*|\":[]/'";
+  private static final String ILLEGAL_NAME_CHARACTERS = "*|\":,[]/'";
```
Comment (thomasdelhomenie) [inline wiki-service/src/main/java/org/exoplatform/wiki/utils/Utils.java]: Is the character , causes issue ? I can upload an attachment containing a ,

File: wiki-service/src/main/java/org/exoplatform/wiki/utils/Utils.java [hunk 2]
```diff
@@ -108,6 +108,7 @@ public static String escapeIllegalCharacterInName(String name) {
       for (char c : ILLEGAL_NAME_CHARACTERS.toCharArray())
         name = name.replace(c, '_');
       name = name.replace("%20", "_");
+      name = name.replace("%22", "_");
```
Comment (thomasdelhomenie) [inline wiki-service/src/main/java/org/exoplatform/wiki/utils/Utils.java]: Why should we do that for double quotes and not for single quote for example ?
The management of special characters in wiki page and attachment names looks a bit ugly. We deal with encoded and non-encoded characters in this method for example. Can we improve that ?

File: wiki-service/src/main/java/org/exoplatform/wiki/utils/Utils.java (hunk 1 above)
Comment (trongtt) [reply wiki-service/src/main/java/org/exoplatform/wiki/utils/Utils.java]: I find there are two different problems. This one is to fix the displaying of the image with special character after uploading.

Comment (trongtt) [reply wiki-service/src/main/java/org/exoplatform/wiki/utils/Utils.java]: And I see this undisplayed image problem with a "," only happens on Chrome, it is working well on Firefox. In conclusion, it is different in handling/accepting filename between browsers indeed.

File: wiki-service/src/main/java/org/exoplatform/wiki/utils/Utils.java (hunk 2 above)
Comment (trongtt) [reply wiki-service/src/main/java/org/exoplatform/wiki/utils/Utils.java]: we do that for double quote only as the file upload is URL encoding it in filename in Chrome (refer https://bugs.chromium.org/p/chromium/issues/detail?id=81193)

I just realized that the double quote in filename is still buggy in Firefox. In Firefox, a " is escaped to \\" which also cause the error message.

I will take time to investigate more about the issue and try to improve this part.

**Comment by trongtt** [comment]
I fixed the issue with "~!@#$%^*()_+`-=[]|}{';:",.<>?.png". It is because of "^" character in the name which is being used for identifying Confluence Syntax.

**Comment by thomasdelhomenie** [comment]
It seems better but I still have issues (when using a file named ~!@#$%25^%2A()_+`-=%7C}{';:",.<>?.png):
* when I add the image in the rich text editor, I only see the image name, not the image
* when I save the page, the image is well displayed but with some characters aside :
selection_048

**Pull request merged**